controller.go

```diff
@@ -301,17 +333,25 @@ func (c *controller) addNetwork(n *network) error {
 	}
 
 	n.Lock()
+	n.ctrlr = c
+	n.endpoints = endpointTable{}
 	n.svcRecords = svcMap{}
 	n.driver = dd.driver
-	d := n.driver
+	n.dataScope = dd.capability.DataScope
 	n.Unlock()
+	return nil
+}
+
+func (c *controller) addNetwork(n *network) error {
 
 	// Create the network
-	if err := d.CreateNetwork(n.id, n.generic); err != nil {
+	if err := n.driver.CreateNetwork(n.id, n.generic); err != nil {
 		return err
 	}
-	if err := n.watchEndpoints(); err != nil {
-		return err
+	if n.isGlobalScoped() {
```

Comment:
Added a check here. I think it's necessary because both local/global network addition will invoke this function.